packages/xdl/src/ProjectAssets.ts

```diff
@@ -414,11 +422,13 @@ export async function resolveAndCollectExpoUpdatesManifestAssets(
       const contents = await fs.readFile(absolutePath);
       // Expo Updates spec dictates that this hash is sha256
       const hash = crypto.createHash('sha256').update(contents).digest('hex');
+      const contentType = mime.getType(absolutePath) ?? 'application/octet-stream';
       manifestAssets.push({
         url: urlResolver(assetPath),
         hash,
         key: assetPath,
-        contentType: mime.getType(absolutePath) ?? 'application/octet-stream',
+        contentType,
+        fileExtension: `.${mime.getExtension(contentType) ?? ''}`,
```

Comment:
Use the file extension from the filename in `absolutePath` to mimic the server's logic. This way we honor the extension used on disk.

If there is no file extension, omit the property (just use undefined here I think). If the file ends in ".", the file extension should be just ".". A string that does not begin with "." is invalid here and the client will prefix it with ".", so an empty string will be treated as if "." were specified. https://docs.expo.dev/technical-specs/expo-updates-0/#manifest-response-headers